code/controllers/subsystem/job.dm

```diff
@@ -164,9 +172,18 @@ SUBSYSTEM_DEF(job)
 			continue
 
 		if((job.current_positions < job.spawn_positions) || job.spawn_positions == -1)
-			JobDebug("GRJ Random job given, Player: [player], Job: [job]")
-			if(AssignRole(player, job.title))
-				return TRUE
+			JobDebug("GRJ Random job can give, Player: [player], Job: [job]")
+			if(CanAssignRole(player, job.title))
```

Comment:
Duplicate check, the stuff CanAssignRole checks for is already checked for above

Reply:
CanAssignRole() is required here as it's looping through each of the possible jobs. If a job if gets isn't assignable for other reasons, it'll still look for the next job in the list.

Without this check, it could common send back a Job that couldn't be assigned and the button for joining as a 'Random Job' could possibly fail a couple of times for no reason (e.g. faulty)